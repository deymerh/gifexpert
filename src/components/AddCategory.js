import React, { useState } from 'react';
import propTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleChange = (e) => {
    setInputValue(e.target.value)
  }
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 3) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('')
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        onChange={handleChange}
        type="text" />
    </form>
  )
}
AddCategory.propTypes = {
  setCategories: propTypes.func.isRequired
}