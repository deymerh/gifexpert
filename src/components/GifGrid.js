import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { loading, data: images } = useFetchGifs(category);

  return loading ? <p>Cargando...</p> : (
    <>
      <h3> {category} </h3>
      <div className="card-grid animate__animated animate__fadeIn">
        {
          images.map(img => {
            return (
              <GifGridItem key={img.id} {...img} />
            )
          })
        }
      </div>
    </>
  )
}
