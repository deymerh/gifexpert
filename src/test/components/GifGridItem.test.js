import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el compoente <GifGridItem  />', () => {

  const url = 'https://images.com/12345';
  const title = 'Goku';
  const wrapper = shallow(<GifGridItem url={url} title={title} />);

  test('Debe mostrar el conponente <GifGridItem /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe mostrar el title correctamente', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });
  test('debe mostrar las propiedades de la img correctamente', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });
  test('debe de tener el findIn en la clase', () => {
    const div = wrapper.find('div').prop('className').includes('animate__bounce');
    expect(div).toBe(true);
  });
});