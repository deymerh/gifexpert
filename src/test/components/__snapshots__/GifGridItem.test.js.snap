// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el compoente <GifGridItem  /> Debe mostrar el conponente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Goku"
    src="https://images.com/12345"
  />
  <p>
    Goku
  </p>
</div>
`;
